services/src/main/java/io/scalecube/services/IServiceRegistry.java

```diff
@@ -0,0 +1,26 @@
+package io.scalecube.services;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Service registry interface provides API to register/unregister services in the system and make services lookup by
+ * service result.
+ */
+public interface IServiceRegistry {
+
+  void registerService(Object serviceObject);
+
+  void unregisterService(Object serviceObject);
+
+  List<ServiceInstance> serviceLookup(String serviceName);
+
+  ServiceInstance serviceInstance(ServiceReference reference);
```

Comment:
What will be returned if reference would not be found? Null?
Better than return optional here. Also it would be good to add javadocs for this interface